### ISSUE TITLE
Let me know if this actually fixes stuttering for you

### ISSUE THREAD

**Issue by soupstream** [soupstream/EldenRingStutterFix#4]
I've been seeing a lot of reports on Discord/Reddit/Nexus that this mod doesn't work and only a few that say it does help. I'd like to figure out whether it actually does fix anything, and in which circumstances it can be expected to help.

If this mod has improved stuttering issues for you, please post your specs here and describe the improvement you've seen. Make sure it wasn't just disabling EAC that improved performance. Any performance numbers you can provide would also be welcome.

**Comment by T4keD0wN** [comment]
Thank you, i have experienced very drastic stuttering before with EAC disabled. Used this for 3h so far and it seems to work, stutter is pretty much either gone or unnoticeable now. 
Specs: 3060ti, 12400, 16gb ram, 970 evo plus, Windows 10 build 19044.1586

**Comment by jbyron1** [comment]
I've still gotten some dips from 60 to like 50, but the dips are way less frequent and way smaller in size than when I tried with just EAC disabled.

This is with a 3070 and a i7-10700KF, windows 19044.1526

**Comment by NickEtlis** [comment]
Still white screen then crashes

**Comment by Tarros78** [comment]
for me no stutter at all and seems  slow down now have minor impact during gamplay...seems shader cache working bettern and faster...notice some pop up in liurna but i just arrived thre today so i don't know if it related to fix...i will try..
gaming for about 3  hour 
spec i7 4790 16 gb ddr 3 1070 1 tb ssd  870

**Comment by jullebarge** [comment]
I had a lot of stutters and with this patch/mod they are still present but less frequent. Thanks a lot. My setup is Ryzen 2600 RTX 3060 Ti 16 Gb DDR3 Windows 11 at 1440p max settings

**Comment by jordevorstenbosch** [comment]
This helped a lot, basically went from constant three in a row 'ultra-slow-mo-to-rapid-catch-up' frame timings to an occasional stutter here and there.

Specs:
- 5800X
- 6900XT
- 32@3200 DDR4
- installed on a current gen NVME SSD

On Windows 10 (19044)

**Comment by RobertBanner** [comment]
Now runs like smooth butter!!

From 20 to 30 fps drops now only 2 fps drops max

Thank you.

**Comment by cmhamiche** [comment]
It feels smoother but it's hard to quantify honestly.

GPU: NVIDIA GeForce GTX 1060 6GB, 
CPU: Intel(R) Core(TM) i5-3470 CPU @ 3.20GHz, 
Resolution: 1920x1080, 
OS: Microsoft Windows 11 Pro 2009 10.0.22000,
GPU Driver : NVIDIA: 511.79
System RAM: 8GB DDR3 1400 MT/s
Sata SSD
No overlays, no EAC, no mods. Vsync disabled in nvidia panel. 
I tried to get some numbers but it's not really useful.

1st run: The First Step to the Church of Elleh straight on foot.
		Avg FPS	Min FPS	Max FPS	90th %	95th %	99th %	Time (ms)
Vanilla :	37.587	16.753	48.325	34.137	32.207	22.443	46.084			
Modded : 38.284	16.728	46.113	35.389	34.129	18.097	42.972

2nd run: Agheel Lake North to the east bridge straight on Torrent.
		Avg FPS	Min FPS	Max FPS	90th %	95th %	99th %	Time (ms)
Vanilla :	39.792	14.354	56.943	37.615	36.546	29.262	34.204
Modded : 38.713	14.197	58.321	36.380	35.240	28.830	34.486

3rd run: The First Step making two back and forth between the site First Step and the crucified on Torrent while moving the mouse left and right (where the Tree Sentinel were).
		Avg FPS	Min FPS	Max FPS	90th %	95th %	99th %	Time (ms)
Vanilla :	35.862	12.825	46.391	31.714	27.561	23.802	43.501
Modded : 35.946	13.714	48.722	32.432	29.174	24.694	31.382

**Comment by immi4146** [comment]
This actually seems to be working, with a Caveat. When I Just Copy/Paste the d3d12 dll into my game directory It didn't seem to do anything. But If I place it in a mods directory using the Modloader made by TechieW on Nexus Mods (https://www.nexusmods.com/eldenring/mods/117) It seems to have actually loaded and changed something now.

I've been Testing (not particularly Scientifically) in the First Step and fighting the Tree Guardian. Previously this area would be a stuttery mess, now it still has some significant drops but its no where near as horrible.

I'm not convinced this is a perfect fix, but it seems to be helping some

GPU: RTX 3090
CPU: R9-5950X
RAM: 32GB 3600Mhz
Resolution: 3440x1440
Settings: Max
OS: Win 10 21H1
NVMe SSD

**Comment by dmpunks** [comment]
I had the invisibile npcs and enemies glitch which was fixed by https://www.nexusmods.com/eldenring/mods/109 but installing this made the enemies disappear again.

**Comment by Lerzu** [comment]
The first time shader generation might be the same but the slowdowns/freezing from hitting load-triggers has been improved with this. I use Stormhill Shack for my testing as running up the hill reduces FPS from 60 to 30 for several seconds and then running back down freezes the game completely for a moment. With this fix, the uphill FPS only drops by 5 FPS and the downhill freezing seems slightly shorter. I also tested the stairs at the beginning of the game, next to the tutorial pit, and the FPS is smoother there too. In short, it's still not perfect but much better than without the fix.

GPU: 6800XT
CPU: 5800X
RAM: 32GB 3200Mhz 16-18-18 Dual rank (was 16GB single rank few days ago but the upgrade did nothing for the stutter, in case anyone was wondering about that)
Resolution: 3200x1800 (makes no difference what resolution I play at, unless it's 8K, which turns the game into a slideshow, as expected)
Settings: custom (DOF and motion blur off, everything else High or Max.
OS: Win10 21H1
SATA 3 SSD

**Comment by TamaZero1** [comment]
Actually worked, to test it i travel to the first zone, where the tree sentinel is, and i didn't see any stuttering at all, i traveled a bit, and i see a little drop, but nothing compared to the EAC one, fix= 60-50fps, EAC 60-10fps, thank you.

GPU:3060
CPU: Ryzen 5 5600X
RAM:16GB
RESOLUTION: 1080
OS: W11
Settings: Everything max except the grass quality, on medium.
SSD SATA3

**Comment by jullebarge** [comment]
Since the new path, it doesn't work anymore on my config.

**Comment by valentyn-l** [comment]
It helped a little, but the stuttering is still present for me in all the egregious areas as before.

GPU: RTX 2080 Ti
CPU: i7 7700K
RAM: 32GB
RESOLUTION: 1800p
OS: W11
Settings: MAX
8TB SSD SATA3 RAID
1TB NVME

**Comment by uOWooLab-JunLuo** [comment]
I have not applied the fixes yet, since it seems to be not working for patch 1.03 which just launched today. However, may I suggest that we figure out a specific area for testing the framerate, so that people can see the effectiveness on different hardware.

**Comment by cmhamiche** [comment]
GPU: NVIDIA GeForce GTX 1060 6GB,
CPU: Intel(R) Core(TM) i5-3470 CPU @ 3.20GHz,
Resolution: 1920x1080,
OS: Microsoft Windows 11 Pro 2009 10.0.22000,
GPU Driver : NVIDIA: 511.79
System RAM: 8GB DDR3 1400 MT/s
Sata SSD
No overlays, no EAC, no mods. Vsync disabled in nvidia panel, shader cache set to unlimited.

The First Step making two back and forth between the site and the crucified poles on Torrent while moving the mouse left and right (at the Tree Sentinel location).


```
Runtime			Avg FPS	Min FPS	Max FPS	90th %	95th %	99th %	Time (ms)	Latency (ms)	GPU0Clk(MHz)	GPU0 Util%	CPUClk(MHz)	CPU Util %
Vanilla 1.0.3		35.990	13.435	50.570	31.201	26.752	22.665	42.040	1.668			1992.976		55.425	3383.724	75.298
EldenRingStutterFix	36.217	13.272	52.622	31.551	27.920	24.062	38.270	1.772			1993.809		54.712	3383.749	74.670
VKD3D-Proton		34.433	10.021	53.954	29.984	28.579	23.241	41.298	22.805		1975.242		93.826	3383.999	75.302
```

dxvk.conf
[eldenring.exe]
PROTON_HIDE_NVIDIA_GPU = 0 
PROTON_ENABLE_NVAPI = 1

**Comment by dedhalo** [comment]
When I run the game, the game told me that there was an error in the application (0xc0000142) and untrusted file "d3d12.dll".

**Comment by DevTechProfile** [comment]
Yes, it helped indeed. Thanks for this great work.

https://twitter.com/CapFrameX/status/1504195757909413893

**Comment by cmhamiche** [comment]
Here's a chart going from The First Step to the forest.
![chart](https://user-images.githubusercontent.com/32405217/158895887-8f1da380-54e4-42ed-b0d1-3827c137b7bb.png)

"MsBetweenPresents (Rendered FPS) should be used to chart the average rendered FPS. This data is captured from the beginning of the graphics pipeline and indicates the smoothness of the animation delivered to the GPU. "

Here's the location:

![location](https://user-images.githubusercontent.com/32405217/158896404-28d364bf-e1cf-47a9-9245-f3222ae19d8b.png)

**Comment by Devologger** [comment]
Does it work with Special K?

**Comment by Dullnets** [comment]
Now it works much better. There are still micro stutter but a lot less and especially less frequent. Thankssssss.

ryzen 5 2600x
16gb ram 3200
rx 6500xt
1080p med/high

**Comment by Pankeko** [comment]
Got a 5-7 fps increase and really reduced stutters, thank you so much, went from 40 to 46 in some areas 

I5 4690
Gtx 1060
16gb ram

**Comment by Jivko1212** [comment]
I found a way to stop the stuttering on my system. If I play with keyboard and mouse I get no stuttering (I was using this fix along with the Elden Ring Frame Rate Unlocker). As soon as I turn on my controller, the game starts to stutter (it pauses two times next to each other for about half a second). I had a similar issue with Sekiro, which was rectified by Special K.

**Comment by Jivko1212** [comment]
There is a discussion thread about the issue here: https://www.reddit.com/r/Eldenring/comments/t47v1l/i_understand_there_are_other_people_getting/
Controller stutter is atrocious. Without it I am getting really good performance with your fix and the frame rate unlocker (70ish fps). Vanilla with EAC is running sub 60 on my 3080, 5900x system and not smooth.

**Comment by Ryusennin** [comment]
Maybe we should agree on what "stuttering" actually covers.

There are multiple issues of stuttering in Elden Ring. But the causes are different. Stuttering due to bad performance (for example 50 fps on a 60 Hz screen), due to controller/mouse/keyboard, due to shader caching (which this mod seems to be about).

The worst offender is the slowmo/accel stuttering which happens everywhere there is a zone transition (like the Tree Sentinel, or the staircase next to the glowing tree in the initial Stranded Graveyard). What happens in those places is that the game will forcibly slow down to 30 fps for a second or two in order to shuffle its assets from RAM to VRAM. This is *not* related to HDD/SSD. The only way right now to fix that particular stuttering is to cap the whole game at 30 fps.

**Comment by Jivko1212** [comment]
I noticed something pretty interesting. If I play in windowed mode and open device manager to the side, every time the game stutters for me device manager refreshes. It would be interesting if other people are also having the same problem.

**Comment by DovaNuts** [comment]
For my this was a day and night, from drops to 5 fps every 30 seconds to almost none or very little to 40, thanks a lot, finally I can enjoy Elden Ring 3 weeks later.

My specs
RX 580 8GB
16GB Ram 3333mhz
Ryzen 2700
Amd driver 22.3.1
Windows 10 Pro 21H2 1904
512GB NVME
Borderless window
Elden Ring 1.02.3
![image](https://user-images.githubusercontent.com/61672375/159151298-753b7dcc-2c75-4c6e-84f0-7c62652eb382.png)

**Comment by FastiJan** [comment]
Constant stutter totally gone while rotating camera around and locked at 60fps instead of 40-50 before.
4930k (4.4GHz), GTX 1080.

In this location near start:
![image](https://user-images.githubusercontent.com/16276877/159153552-b6408716-5df5-4a65-9ae4-83a9224ed4b2.png)

**Comment by DasUnding** [comment]
It works for me no more stuttering by hitting etc. thank you for it great stuff

**Comment by rabidcopy** [comment]
Went from hitching and frametime spikes with no "perceivable" reason every 10-20 seconds to virtually no fluctuations in framepacing or hitching. Ryzen 2600 + RX 570 + 16GB 3600Mhz RAM. 1600x900 with low-ish settings with high shadows and antialiasing. Has been a definite game changer for me. Beats running the game on Linux through Proton/VKD3D for these fixes because my RX 570 can just barely run the game at 60 FPS to begin with and I lose about 15-20 FPS on Linux.

**Comment by sam86314** [comment]
Does seem to work on my main system, though I haven't had too many issues with stuttering.

Specs: R7-5800X (PBO enabled), 32GB DDR4-3600 C18, RX 5700XT (underclocked to 1850MHz @950mV), 3440x1440 max settings

The game is running off of a "RAID-0" SATA SSD array, and EAC is removed.

I'll probably test it on my i5-1135G7 laptop at some point. I think that had more issues with stuttering.

**Comment by ProgTheater** [comment]
work for me with ryzen 2600 and RX 590 8gb, I'm wondering if this is compatible with the last version of the game 1.03.1, I'm currently on 1.02.3

**Comment by leechgrrl** [comment]
Completely fixed the constant stuttering I was having.

Specs:
Ryzen 3600 + Radeon 6700 XT
Windows 11 latest dev build

**Comment by Spark9999** [comment]
So it doesn't fix the stuttering for me, particularly during bosses which have a great number of visual effects all at once like when they use magic or when the giant flowers in Liurnia all launch their magic projectiles at the player and all thos projectiles combined create a lagging/stuttering effect until the magic is dispersed.

Playing on the latest nvdia grd driver 511.79

Specs :
Windows 10 pro 21H1 latest build (20H2 is optional for now)
i5-10400
16gb ram
gtx 1650
EAC is disabled completely during play

**Comment by soupstream** [comment]
Thanks for all the responses. It seems like there are multiple sources of stuttering in Elden Ring and this mod fixes one of them for some people.

**Comment by rabidcopy** [comment]
While I know this issue is closed and further maintenance of this project is over. I want to say in my own testing, I still get random stutters when using Special K versus the DLL provided here. I scoured the Special K Discord and saw several posts from the developer claiming Special K already does everything these patches do, but there were some conflicting statements where the developer claimed these patches ported over to Windows D3D12 with this project specifically shouldn't do anything and will only affect performance through VKD3D on Linux. I still get less stutter over vanilla when using Special K, but approximately every 10-20 seconds I will get a stutter/hitch, even when standing still and not moving the camera. This does not occur at all for me when using EldenRingStutterFix. 
![Untitled](https://user-images.githubusercontent.com/8052832/159750109-4f4303f0-a6eb-4528-89fb-2f15d3e7249b.png)
I don't see any headway being made approaching the developer on this claim with my own personal anecdote. It would be quickly deemed a placebo, a margin of error, or some form of user error.

tl;dr until From Software fixes their game, this will continue to be my go-to for making the game far much more bearable to play

**Comment by Genkishi569** [comment]
Hi, I'm here about the news that you've decided not to maintain this because SpecialK implements the same fixes. SpecialK also crashes you when it detects you invading an NPC (you know...those Volcano Manor invasion quests). It kept playing the MGS alert sound then crashed me every single time I tried to invade Tragoth (patches' questline). It's incredibly annoying and heavy-handed, imo.

Additionally, SpecialK actually made things worse for me (tried if before your fix...tried it again after your announcement). I've never once dropped below 40fps...but with specialK...my fps counter basically dips into low 40s and low 30s which has never happened before. Your fix is simple and does wonders for me, and is simple to install and remove.

Lastly, SpecialK's handling of the borderless window mode is incredibly weird. It actually feels like I'm running semi-exclusive fullscreen with weird flickers everytime I alt+tab in or out...whereas elden ring's default borderless is just fluid, and works the way I expect borderless should.

All I'm saying is that I'm sad to see that you won't be maintaining this fix even though it's a lot easier and more painless to use compared to SpecialK. I'd love it if you saw this and re-considered maybe maintaining the fix for awhile longer because it's pretty damn painless to use, and can actually be used online...which is where reducing stuttering would actually be an incredible boon.

**Comment by Patrice667** [comment]
Hello Soupstream,

I have tested the .dll fix again and it works this time. I did something wrong in my game before.
Good work, stutters disapear for 80% and when it happens, frametimes are much shorter.
I hope this file can run when the game will be patched beyond v1.03.x
Will multiplayer be possible with this fix in the future?
Anyway, good work!

Kind regards,

Patrice Beuger
